src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -170,6 +174,20 @@ public IStringLocalizer Create(string baseName, string location)
             });
         }
 
+        protected virtual string GetResourcePrefix(string location, string baseName, string resourceLocation)
+        {
+            // Re-root the base name if a resources path is set
+            return location + "." + resourceLocation + TrimPrefix(baseName, location + ".");
+        }
+        private string GetResourcePath(Assembly assembly)
+        {
+            var resourceLocationAttribute = assembly.GetCustomAttribute<ResourceLocationAttribute>();
+            // If we don't have an attribute assume all assemblies use the same resource location.
+            return string.IsNullOrEmpty(resourceLocationAttribute?.ResourceLocation)
```

Comment:
`ResourceLocation` can't be `null` if the attribute exists. Just check `resourceLocationAttribute == null`.